torch/csrc/distributed/autograd/rpc_messages/rpc_with_autograd.cpp

```diff
@@ -53,27 +56,30 @@ Message RpcWithAutograd::toMessage() && {
   auto payload = std::move(wrappedMessage_).movePayload();
   TORCH_INTERNAL_ASSERT(!payload.empty());
 
-  // We append the message type (1 byte), autograd context id(8 bytes) and
-  // autograd message id(8 bytes) to the original message in network byte order
-  // (big endian).
-  size_t writableIndex = payload.size();
+  std::vector<at::IValue> ivalues{messageType,
+                                  autogradMetadata_.autogradContextId,
+                                  autogradMetadata_.autogradMessageId,
+                                  fromWorkerId_};
 
-  // Need 17 additional bytes.
-  payload.resize(payload.size() + kAutogradMessageSize);
+  // Now pickle using JIT pickler.
+  std::vector<torch::Tensor> tensorTable;
+  std::vector<char> additionalPayload =
+      jit::pickle(c10::ivalue::Tuple::create(std::move(ivalues)), &tensorTable);
 
-  // Add message type.
-  payload[writableIndex++] = messageType;
+  // We shouldn't have any tensors!
+  TORCH_INTERNAL_ASSERT(tensorTable.empty());
 
-  // Add autograd ids.
-  torch::utils::THP_encodeInt64Buffer(
-      reinterpret_cast<uint8_t*>(payload.data()) + writableIndex,
-      &autogradMetadata_.autogradContextId,
-      torch::utils::THPByteOrder::THP_BIG_ENDIAN,
-      1);
-  writableIndex += sizeof(int64_t);
+  // Append the payload.
+  payload.insert(
```

Comment:
Is the additional payload for autograd information going to be a non-fixed size now, as opposed to 17 before? Should we also keep the comment about what is appended to the message for clarity/understanding?

Reply:
Its probably fixed size now, but might change later if we decide to add/remove things from the IValue tuple. I had the comment originally since I was manually serializing the bytes. I think for now the code clearly outlines what we are serializing using `jit::pickle`.